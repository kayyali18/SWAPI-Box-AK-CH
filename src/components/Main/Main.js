import React, { Component } from "react"
import PropTypes from "prop-types"

import NavBar from "../NavBar/NavBar"
import CardContainer from "../CardContainer/CardContainer"
import Card from "../Card/Card"


class Main extends Component {
  constructor() {
    super()

    this.state = {
      planets: null,
      people: null,
      vehicles: null,
      category: null,
      cards: null
    }
  }

  componentDidMount = () => {
    const { people, planets, vehicles } = this.props
    this.setState({
      planets,
      people,
      vehicles
    })
  }

  render() {
    const {generateCards} = this.props
    return (
      <main className="l-main">
        <NavBar generateCards={generateCards}/>
        <CardContainer
          cards={this.state.cards}
          category={this.state.category}
        />
      </main>
    )
  }
}

export default Main

Main.propTypes = {
  people: PropTypes.object,
  planets: PropTypes.object,
  vehicles: PropTypes.object,
  generateCards: PropTypes.func
}
