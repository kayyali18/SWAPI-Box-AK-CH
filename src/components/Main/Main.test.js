import React from "react"
import ReactDOM from "react-dom"
import Main from "./Main"
import { shallow } from "enzyme"

describe("App", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Main />)
  })

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should update state to not match snapshot", () => {

  })

  it("should generate cards on click", () => {

  })
})