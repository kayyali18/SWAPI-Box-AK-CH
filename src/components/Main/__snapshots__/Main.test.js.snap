// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<main
  className="l-main"
>
  <NavBar
    generateCards={[Function]}
  />
  <CardContainer
    cards={null}
    category={null}
  />
</main>
`;
